src/analysis/retail/mage/arcane/talents/RadiantSpark.tsx

```diff
@@ -1,79 +1,136 @@
+    this.active = this.selectedCombatant.hasTalent(TALENTS.RADIANT_SPARK_TALENT);
     this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(CAST_SPELLS), this.onCast);
     this.addEventListener(
       Events.removebuff.by(SELECTED_PLAYER).spell(TALENTS.RADIANT_SPARK_TALENT),
       this.onRadiantSparkRemoved,
     );
   }
 
+  finishRampPhase = () => {
+    this.arcaneBlastCasts = 0;
+    this.arcaneSurgeCasts = 0;
+    this.aoeCasts = 0;
+    this.hasRuneOfPowerBuff = true;
+  };
+
   onCast(event: CastEvent) {
     //If Radiant Spark is not active, then we do not need to check the Arcane Blast cast.
     if (!this.selectedCombatant.hasBuff(TALENTS.RADIANT_SPARK_TALENT.id)) {
       return;
     }
 
+    //If we dont have Rune of Power on any cast, flag it as false
+    if (!this.selectedCombatant.hasBuff(SPELLS.RUNE_OF_POWER_BUFF.id)) {
+      this.hasRuneOfPowerBuff = false;
+    }
+
     const spellId = event.ability.guid;
+
     if (spellId === SPELLS.ARCANE_BLAST.id) {
       this.arcaneBlastCasts += 1;
+    } else if (spellId === TALENTS.ARCANE_SURGE_TALENT.id) {
+      this.arcaneSurgeCasts += 1;
     } else if (getHitCount(event) >= AOE_TARGET_THRESHOLD) {
       this.aoeCasts += 1;
     }
   }
 
   onRadiantSparkRemoved(event: RemoveBuffEvent) {
-    //Confirm whether 5 Arcane Blast casts (4 with Harmonic Echo) were cast during Radiant Spark
-    const spellCountThreshold = this.hasHarmonicEcho
-      ? CASTS_PER_RADIANT_SPARK - 1
-      : CASTS_PER_RADIANT_SPARK;
-    if (this.arcaneBlastCasts !== spellCountThreshold && this.aoeCasts < spellCountThreshold) {
+    /* All Cases - Must have Rune of Power Buff */
+    /* Case 1 - Burn Phase. Need to cast 3 Blasts + Surge */
+    /* Case 2 - Mini Burn. No AS. Need to cast 4 Blasts */
+
+    let wasBadRadiantSpark = false;
+
+    if (!this.hasRuneOfPowerBuff) {
+      debug && this.log('You are not standing in Rune of Power when doing a Radiant Spark Ramp.');
+
+      wasBadRadiantSpark = true;
+      this.badRadiantSparkBecauseOfRune += 1;
+    }
+
+    if (this.arcaneSurgeCasts === 1 && this.arcaneBlastCasts < 3) {
+      debug &&
+        this.log(
+          'You didnt cast 3 Arcane Blasts before Arcane Surge during the Radiant Spark Ramp. Casted ' +
+            this.arcaneBlastCasts +
+            ', expected 3',
+        );
+
+      wasBadRadiantSpark = true;
+      this.badRadiantSparkBecauseDidntUseFourSpells += 1;
+    } else if (this.arcaneSurgeCasts === 0 && this.arcaneBlastCasts < 4) {
       debug &&
         this.log(
-          'Not enough (or too many) Arcane Blast casts during Radiant Spark - Casted ' +
+          'You didnt cast 4 Arcane Blasts during the Radiant Spark Ramp. Casted ' +
             this.arcaneBlastCasts +
-            ', expected ' +
-            spellCountThreshold,
+            ', expected 4',
         );
+
+      wasBadRadiantSpark = true;
+      this.badRadiantSparkBecauseDidntUseFourSpells += 1;
```

Comment:
Will fix the conflicts and come back with a new commit.

Regarding the combine the if statement into surge + blast < 4 - I had taken the decision to do this in 2 steps because I wanted to code to be readable to a future developer in a way that it should clearly match the rules I have described and am trying to meet. So it would be like

- Rule 1 - 1 Surge and 3 Blasts - ok this code block is for this rule.
- Rule 2 - 4 Blasts - the next block is for this rule.

If you still want me to change it, I will. Maybe in future modules I can come up with a better 'rule checker' structure instead of chained if/elses.